src/node_file.cc

```diff
@@ -3431,6 +3432,27 @@ static void CpSyncCopyDir(const FunctionCallbackInfo<Value>& args) {
   bool verbatim_symlinks = args[5]->IsTrue();
   bool preserve_timestamps = args[6]->IsTrue();
 
+  std::function<bool(std::string, std::string)> filter_fn = nullptr;
+
+  if (args[7]->IsFunction()) {
+    Local<v8::Function> args_filter_fn = args[7].As<v8::Function>();
+
+    filter_fn = [env, args_filter_fn](std::string src,
+                                      std::string dest) -> bool {
+      Local<Value> argv[] = {
+          String::NewFromUtf8(
+              env->isolate(), src.c_str(), v8::NewStringType::kNormal)
+              .ToLocalChecked(),
```

Comment:
Can we avoid tolocalchecked() calls since they are unsafe. Also you can use ToV8Value method which reduces the code.

Reply:
Regarding the safeness of the code, I did run this with different filter functions, I also tried not returning anything or throwing within the filter function itself and every case seems totally fine with `ToLocalChecked` :thinking:

Reply:
The key challenge with `ToLocalChecked()` is that it causes the node.js process to crash even on otherwise recoverable errors. It really is better to use the `ToLocal(...)` APIs instead and we've been going through and actively converting away from using `ToLocalChecked` as much as possible. I agree with @anonrig that we should avoid it here also.

Reply:
If string creation fails due to external factors or simply due a bug in v8, this will directly crash process

Reply:
Ok, thanks a lot for the clarification both :heart_hands:, I'll swap this with `ToLocal` :+1: